R/parse.R

```diff
@@ -321,18 +321,14 @@ prof_sort <- function(prof) {
   # stack.
   prof_split <- vctrs::vec_split(prof, prof$time)$val
 
-  max_depth <- max(map_int(prof_split, nrow))
+  max_depth <- max(vapply(prof_split, nrow, integer(1)))
   n_samples <- length(prof_split)
 
-  # Extract labels (function names for the stack frames) and pad
-  # them with missing values so they are of equal lengths
+  # Extract labels (function names for the stack frames) and pad with NAs
+  # so we can easily make a data frame
   pad <- function(x) x[seq_len(max_depth)]
-  stacks <- map(prof_split, function(x) pad(rev(x$label)))
-
-  # Transpose into a data frame. The (unnamed) columns represent
-  # increasing depths in the stacks.
-  stacks <- map(transpose(stacks), simplify)
-  stacks <- vctrs::data_frame(!!!stacks, .name_repair = "minimal")
+  stacks <- lapply(prof_split, function(x) pad(rev(x$label)))
+  stacks <- as.data.frame(do.call(rbind, stacks))
```

Comment:
We can simplify the existing code because we already have a list of vectors where every vector is the same length. 

Could also do:

```R
stacks <- vapply(prof_split, function(x) pad(rev(x$label)), character(max_depth))
stacks <- as.data.frame(t(stacks))
```

But I don't think that's much of an improvement.